commit bcccc50ce8fcc833cfed4bb71ede211a6ef5b84a
Author: Vincent Sanders <vince@kyllikki.org>
Date:   Tue Mar 13 15:34:17 2012 +0100

    ARM: 7420/1: Improve build environment isolation
    
    Increasingly distributions are setting default build environments to
    have LDFLAGS with hardening options. There seems to be an assumption
    with those options that LDFLAGS are passed to the compiler frontend
    rather than used directly with ld (which the kernel build process
    assumes)
    
    To prevent build failures in such environments this patch changes the
    ARM architecture Makefile to override the LDFLAGS from the environment
    similar to the behaviour on other common architectures e.g. x86
    
    Signed-off-by: Vincent Sanders <vince@collabora.co.uk>
    Signed-off-by: Russell King <rmk+kernel@arm.linux.org.uk>

diff --git a/arch/arm/Makefile b/arch/arm/Makefile
index 0298b00fe241..f8ebf1e97027 100644
--- a/arch/arm/Makefile
+++ b/arch/arm/Makefile
@@ -10,6 +10,9 @@
 #
 # Copyright (C) 1995-2001 by Russell King
 
+# Ensure linker flags are correct
+LDFLAGS		:=
+
 LDFLAGS_vmlinux	:=-p --no-undefined -X
 ifeq ($(CONFIG_CPU_ENDIAN_BE8),y)
 LDFLAGS_vmlinux	+= --be8